ext/node_resolver/resolution.rs

```diff
@@ -221,10 +221,8 @@ impl<TEnv: NodeResolverEnv> NodeResolver<TEnv> {
       specifier,
       referrer,
       referrer_kind,
-      match referrer_kind {
-        NodeModuleKind::Esm => DEFAULT_CONDITIONS,
-        NodeModuleKind::Cjs => REQUIRE_CONDITIONS,
-      },
+      // even though the referrer may be CJS, if we're here that means we're doing ESM resolution
+      DEFAULT_CONDITIONS,
```

Comment:
I introduced this by accident here and we had no tests for it: https://github.com/denoland/deno/pull/24492/files#diff-05634a82e4e5d22732b8ea32675a5422616fc7800d08ada3f5598f85dbf7ce41R216-R225